src/duk_debugger.c

```diff
@@ -149,6 +150,21 @@ DUK_INTERNAL void duk_debug_skip_byte(duk_hthread *thr) {
  *  Debug connection read primitives
  */
 
+/* Peek ahead in the stream one byte. */
+DUK_INTERNAL uint8_t duk_debug_peek_byte(duk_hthread *thr) {
+	/* It is important not to call this if the last byte read was an EOM.
+	 * Reading ahead in this scenario would cause unnecessary blocking if
+	 * another message is not available.
+	 */
+
+	duk_uint8_t x;
+
```

Comment:
By the way, should there be a `DUK_ASSERT(thr->heap->dbg_have_next_byte == 0);` here? In other words, it's not allowed to call `duk_debug_peek_byte()` twice without consuming the pushed back byte first?